src/SerializableClosure.php

```diff
@@ -115,7 +115,12 @@ public function unserialize($serialized)
     {
         // Unserialize the data and reconstruct the SuperClosure.
         $this->data = unserialize($serialized);
-        $this->reconstructClosure();
+        try {
+            $this->reconstructClosure();
+        } catch (\ParseException $e) {
+            // Discard the parse exception, we'll throw a custom one
```

Comment:
Maybe throw a custom exception instead of just a `ClosureUnserializationException`?